dev/run.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python
+"""
+This script can be used to run all things dev. Environment setup, Style-checks, Testing etc.
+"""
+
+from __future__ import print_function
+
+import os
+import subprocess
+import sys
+
+import argcomplete
+import argh
+import six
+
+import pylint as _pylint
+import prospector as _prospector
+import yapf as _yapf
+
+
+# path of directory this file resides in
+PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+HOME = os.getenv('HOME', '')    # HOME environment variable
+
+
+def _list_files_with_extension(dir_name, ext):
+    return [os.path.join(dir_name, f) for f in os.listdir(dir_name) if f.endswith(ext)]
+
+
+def _get_configured_env(_required_env):
+    spark_home = _required_env["SPARK_HOME"]
+    scala_version = _required_env["SCALA_VERSION"]
+
+    if not spark_home:
+        raise ValueError("set SPARK_HOME environment variable to configure environment")
+    if not scala_version:
+        raise ValueError("set SCALA_VERSION environment variable to configure environment")
+
+    configured_env = {}
+
+    spark_lib_path = os.path.join(spark_home, "python/lib")
+    configured_env["LIBS"] = ":".join(_list_files_with_extension(spark_lib_path, ".zip"))
+
+    scala_version_major_minor = ".".join(scala_version.split(".")[0:2])
+    assembly_path = os.path.join(PROJECT_DIR, "target/scala-" + scala_version_major_minor)
+    jar_path = ":".join(_list_files_with_extension(assembly_path, ".jar"))
+    configured_env["JAR_PATH"] = jar_path
+
+    python_paths = [
+        os.getenv("PYTHONPATH", ""),
+        os.path.join(PROJECT_DIR, "python"),
+        _list_files_with_extension(assembly_path, ".jar")[-1],
+        os.path.join(spark_home, "python"),
+        configured_env["LIBS"]
+    ]
+    configured_env["PYTHONPATH"] = ":".join(python_paths)
+
+    return configured_env
+
+
+def configure_env():
+    current_env = dict(os.environ)
+    configured_env = _get_configured_env(current_env)
+    for k, v in configured_env.items():
+        if six.PY2:
+            os.environ[k] = v
+        else:
+            os.environ.putenv(k, v)
+
+
+def call_subprocess(process, keyword_args, trail_args):
+    """wrapper function to format kwargs into bash arguments, print and then make the call"""
+    single_char_options = [k for k, v in keyword_args.items() if v and len(k) == 1]
+    multiple_char_options = [k for k, v in keyword_args.items() if v and len(k) > 1]
+    opts = ["-{}{}".format(k.replace("_", "-"), keyword_args[k]) for k in single_char_options]
+    opts += ["--{}={}".format(k.replace("_", "-"), keyword_args[k]) for k in multiple_char_options]
+    print("calling subprocess: {}".format([process, ] + opts + list(trail_args)))
+    return subprocess.call([process, ] + opts + list(trail_args))
```

Comment:
Can we pass `configure_env` to `subprocess.call` instead of setting the env as a side effect?

Reply:
based on the documented api, it looked like `subprocess.call` did not take that argument, turns out it does. updated!